prow/cmd/deck/static/spyglass/spyglass.ts

```diff
@@ -58,13 +59,13 @@ window.addEventListener('message', async (e) => {
         break;
       case "request": {
         const req = await fetch(urlForLensRequest(lens, index, 'callback'),
-          {body: message.data, method: 'POST'});
+          {body: message.data, method: 'POST', headers: {'X-CSRF-Token': csrfToken}});
```

Comment:
This should specify `credentials: 'same-origin'` in order to ensure the CSRF cookies are actually passed through. While that is the default _now_, it is a sufficiently recent change that we shouldn't rely on it (e.g. Firefox's current ESR version is 60, which predates this change).

```suggestion
          {body: message.data, method: 'POST', headers: {'X-CSRF-Token': csrfToken}, credentials: 'same-origin'});
```

Also, given that these three calls are completely identical, perhaps they should be extracted out to a helper function.